### ISSUE TITLE
ResetTool does not trigger change events on Range1d

### ISSUE THREAD

**Issue by philippjfr** [bokeh/bokeh#6075]
When attaching a callback on the x- and y-range ``start`` and ``end`` atttributes you expect it to be triggered whenever the range changes, including when resetting the range with the reset tool. While this seems to be handled correctly for for ``DataRange1d``, when using explicitly defined ``Range1d`` models no event is triggered as the example below shows:

```python
from bokeh.plotting import figure, output_notebook
from bokeh.models import CustomJS
from bokeh.models import Range1d

output_notebook()

x_range = Range1d(start=0, end=10)
y_range = Range1d(start=0, end=10)

p = figure(
      # Commenting out this line will stop ResetTool triggering the callback 
      # x_range=x_range, y_range=y_range) 
)

cb = CustomJS(code='console.log("Range changed")')
p.x_range.js_on_change('start', cb)
p.x_range.js_on_change('end', cb)
p.y_range.js_on_change('start', cb)
p.y_range.js_on_change('end', cb)

p.circle([1, 2, 3, 4, 5], [2, 5, 8, 2, 7], size=10)

show(p)
```

This also applies to ``on_change`` events on the server.

**Comment by philippjfr** [comment]
@bryevdv Sorry to bug you but it would be a shame if we couldn't fix this for 0.12.5 as it affects all datashader plots. If you could have a quick look at this to tell me what should be happening I'd be happy to work on a PR.

**Comment by mattpap** [comment]
@philippjfr, look into `reset()` in `models/ranges/range1d.coffee`. There's `setv()` call with `silent: true`, which doesn't allow propagation of events. I'm not sure why this is the way it is. There may or may not be a reason.

**Comment by philippjfr** [comment]
>There's setv() call with silent: true, which doesn't allow propagation of events.

I saw that, but DataRange1d appears to do the same thing but still triggers the change events so I think it's something to do with ``update_dataranges``, which only applies to DataRange1d models.

**Comment by mattpap** [comment]
In `DataRange1d`'s case, `start` and `end` are managed elsewhere (in `update()`) and are not guarded with `silent: true`.

**Comment by philippjfr** [comment]
Okay, that makes sense, maybe removing the silent true in Range1d reset is sufficient then, I tried that initially and it seemed to work, but I had assumed there was a reason for it.

**Comment by canavandl** [comment]
The reason for it was to only trigger one change event per property change. But I see we're only triggering a "change" event and not a "change:start" or "change:end" event.

**Comment by mattpap** [comment]
@canavandl, if we want to just support event propagation, then triggering all kinds of change events is not enough, because `silent` blocks any outgoing traffic to server.

**Comment by mattpap** [comment]
This `silent: true` was added quite recently in PR #5253.

**Comment by canavandl** [comment]
I understand that. The issue is/was that the code was setting range properties in multiple places, so that a single drag would issue multiple change events. That's why we made the `setv`'s silent then manually triggered the change. But I understand now how that implementation isn't working for server event handling.

**Comment by mattpap** [comment]
@canavandl, your explanation is what I presumed. I've been dealing with similar issues in layout/rendering recently. @philippjfr, I doubt this is fixable without proper reevaluation of how events are propagated inside bokehjs. I intent to do this as continuation of PR #5976.

**Comment by philippjfr** [comment]
I can see how what @canavandl is a problem when setting ranges in general, but in the case of a Range1d.reset triggered by the ResetTool there is only one place that can happen so does the same issue apply?

**Comment by canavandl** [comment]
resolved by #6082

**Comment by philippjfr** [comment]
Thanks @canavandl!
